app/lib/task/backend.dart

```diff
@@ -328,26 +330,22 @@ class TaskBackend {
     String packageName, {
     bool updateDependants = false,
   }) async {
-    var lastVersionCreated = initialTimestamp;
-    await withRetryTransaction(_db, (tx) async {
-      final pkgKey = _db.emptyKey.append(Package, id: packageName);
+    final package = await packageBackend.lookupPackage(packageName);
+    if (package == null) {
+      return; // assume package was deleted!
+    }
+    final lastVersionCreated = package.lastVersionPublished ?? initialTimestamp;
+    final packageVersions =
+        await packageBackend.listVersionsCached(packageName);
```

Comment:
This defeats the purpose of detecting changes :D

Sure cache, should be up to date, but it might not be.
That's fine when serving API or website, because those only need to eventually be consistent. If we use cache here, we won't eventually be consistent, until a weekly background task runs `backfillTrackingState` -- I think that's a long time to wait.

If the tracking logic crashes for a while, then it's nice that `backfillTrackingState` will eventually ensure consistency.

Reply:
I think version list queries are expensive. Why not use more frequent cheaper status checks than do the costly queries every time? The background task could run daily or like every 8 hours...

Reply:
The cheap thing will usually not return the version that was just published.

`trackPackage` is usually called when we've detected that a package has changed. This is exactly the time at which using the cache will give an outdated result.

Look at `_scanForPackageUpdates` it tracks updates, and only calls `trackPackage` for every change.